manual/manual/tutorials/objectexamples.etex

```diff
@@ -220,7 +245,22 @@ in sections \ref{ss:reference-to-self} and \ref{ss:parameterized-classes}.
 \label{ss:reference-to-self}
 
 A method or an initializer can send messages to self (that is,
-the current object).  For that, self must be explicitly bound, here to
+the current object).
+TODO:
+>> “can send messages to self”  This is confusing.  I think what’s meant is that
+it’s possible to refer to methods and variables that are part of the object.  I’m
+vaguely aware of an object metaphor based on messages, but it seems to me OCaml
+does not actually create messages.
+
+> the use of the word "messages" is largely accurate, I believe.
+
+I'm sure "sending messages" is the commonly used term by some.  But I think it's not
+the term that would be used by Java/C++ developers.  They would think of referencing
```

Comment:
I don't advocate turning this into an "OCaml for \<language> developers".  But the "sending messages" seems quite confusing for someone used to conventional languages.  (To me it suggests creating a buffer, writing data into it, somehow "sending", ...)

I take the statement in the first paragraph of Chapter 1 as a metric for evaluating the contents of Part I of the manual ("This part of the language is a tutorial introduction to the OCaml language.  A good familiarity with programming in a conventional language (say, Pascal or C) is assumed, but no prior exposure to functional languages is required.")  This particular wording seems not to meet that standard--in my opinion.

Surely there is a clearer way to express this idea.  If we can quickly agree on something better, let's go with it.  If not, let's drop the suggestion and move on to things we can agree on.

Reply:
> But the "sending messages" seems quite confusing for someone used to conventional languages

I think that the message metaphore was one of the original explanation for OO (it is used in smalltalk for instance). But since the word is used only once in chapter, you are probably right that it should be paraphrased.

Reply:
Yes, "sending messages" is Smalltalk terminology, but Smalltalk is the birthplace of OOP.  Time for some Alan Kay quotes:

*Actually I made up the term "object-oriented", and I can tell you I did not have C++ in mind.*

*I'm sorry that I long ago coined the term "objects" for this topic because it gets many people to focus on the lesser idea. The big idea is "messaging"*

Reply:
> Yes, "sending messages" is Smalltalk terminology, but Smalltalk is the birthplace of OOP.  
Time for some Alan Kay quotes:

Good quotes, but they recognize that common thinking/usage is not what Kay would prefer.  Innovators put out their best ideas and the world may adapt them (or distort them or ignore them).

How about this?  (And if not, I won't make any change here.)

```
A method or an initializer can send messages to self (that is,
the current object).  (Or, in the terms used in mainstream
languages such as Java and C++, a method or an initializer can call
methods on the current object.)
```

Reply:
I fear that this sounds more like an armistice statement rather than a simple explanation.
I think going to the simpler version,

> A method or an initializer can invoke method on self (that is, the current object).

would be fine. There is definitively merit to try to spread potentially better terminology, but I think that the internal coherency of the chapter is more important here: methods are introduced in the first [section](http://caml.inria.fr/pub/docs/manual-ocaml/objectexamples.html#ss%3Aclasses-and-objects) with 

> We now invoke some methods to p: 

Thus, reusing `invoke` may help readers.